packages/steps/src/BuildStepContext.ts

```diff
@@ -107,7 +108,17 @@ export class BuildStepGlobalContext {
     return this.stepById[stepId].getOutputValueByName(outputId);
   }
 
-  public interpolate(value: string): string {
+  public interpolate<InterpolableType extends string | object>(
+    value: InterpolableType
+  ): InterpolableType {
+    if (typeof value === 'string') {
+      return this.interpolateString(value) as InterpolableType;
+    } else {
+      return this.interpolateObject(value) as InterpolableType;
+    }
+  }
+
+  public interpolateString(value: string): string {
     return interpolateWithGlobalContext(value, (path) => {
```

Comment:
It seems inconsistent to me that we do some of the changes here and some of the changes in the `template.ts` 🤔 

Instead of making the special case for objects here, we should alter the `interpolateWithGlobalContext` function the same way as we did with `interpolateWithOutputs`. So all of our utils functions work both for `objects` and `strings` and not that some do and some don't.

Reply:
What do you think?

Reply:
I agree, moved the logic to `template.ts` for consistency